### PR TITLE
Feature/make uds aux open source

### PR DESCRIPTION
**Pull request by BKaDamien**
Just put the public part of uds aux to kiso-testing

### CODE REVIEW

File: src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+        try:
+            global Uds, createUdsConnection, IsoServices
+            from uds import IsoServices, Uds, createUdsConnection
+
+            Uds = Uds
+            createUdsConnection = createUdsConnection
+            IsoServices = IsoServices
+        except ImportError:
+            log.exception("Could not import python-uds")
```
Comment (sebclrsn) [inline src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py:66]: Out of the scope of this PR, but could you raise instead? It doesn't really make sense to continue if this import fails

File: src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        return super(UdsAuxiliary, cls).__new__(cls)
+
+    def __init__(
+        self,
+        com: CChannel,
+        odx_file_path: str,
```
Comment (sebclrsn) [inline src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py:72]: Can you please make this param optional? It doesn't really make sense to keep it mandatory just to explicitely set it to None

**Review by sebclrsn — APPROVED** [review]

File: src/pykiso/lib/auxiliaries/udsaux/__init__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+TODO
```
Comment (sebclrsn) [inline src/pykiso/lib/auxiliaries/udsaux/__init__.py:11]: Can you please update this?

File: docs/examples/uds_auxiliary.rst [hunk 4]
```diff
@@ -0,0 +1,295 @@
+.. _uds_auxiliary:
+
+Using UDS protocol
+==================
+
+UdsAuxiliary class (:py:class:`pykiso.lib.auxiliaries.udsaux.uds_auxiliary.UdsAuxiliary`) contained
+in uds_auxiliary.py is the main interface between user and all the behind implemented logic.
+This class defines usable keywords(methods) for scripters in order to send a uds request to the device under test (raw or configurable)...
```
Comment (sebastianpfischer) [inline docs/examples/uds_auxiliary.rst:8]:
```suggestion
This class defines usable keywords(methods) for scripters in order to send  uds requests to the device under test (raw or configurable)...
```

File: docs/examples/uds_auxiliary.rst [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+            self.assertFalse(diag_session_response.is_negative)
+
+        def tearDown(self):
+            """Hook method from unittest in order to execute code after test case run.
+            """
+            pass
+
+Send UDS Config Request
+-----------------------
+| Send UDS request as a configurable data dictionary. This method can be more practical for UDS requests with long payloads and a multitude of parameters.
```
Comment (sebastianpfischer) [inline docs/examples/uds_auxiliary.rst:154]: If I remembered well, this method can only be used if the command is defined in the .odx file right. Can we mention it?

File: src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+
+Uds = None
+createUdsConnection = None
+IsoServices = None
+
+
+class UdsAuxiliary(AuxiliaryInterface):
+    """Auxiliary used to handle UDS messages"""
+
+    NEGATIVE_RESPONSE = 0x7F
```
Comment (sebastianpfischer) [inline src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py:47]: Why do we have here the negative response defined when it is already defined in uds_response.py?

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]

File: src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py (hunk 6 above)
Comment (BKaDamien) [reply src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py:47]: I jsut split the UDS aux nothing else

**Review by sebclrsn — APPROVED** [review]

**Review by Pog3k — APPROVED** [review]

**Pull request merged**